### DISCUSSION TITLE
Any plans for proxy_wasm?

### DISCUSSION THREAD

**Discussion by andrewdavidmackenzie** [linkerd/linkerd2#4389] [General] [answered]
I was wondering if there have been any investigations related to adding support for [proxy_wasm](https://github.com/proxy-wasm) in linkerd.

As you will know, rust has a great wasm story and tooling, and it could be a great implementation of it.

The reason I am asking is to gauge interest by different proxy providers for proxy_wasm and hence the perspectives for portable proxy filters/plugins.

thanks

**Comment by olix0r** [comment, marked as answer]
Hi @andrewdavidmackenzie,

We've been following proxy_wasm's progress with great interest. I've recently done some [very early dabbling/prototyping](https://github.com/olix0r/waxy) to start getting a better understanding of WASM and, specifically, how it interacts with an async environment like that of [Tokio](https://tokio.rs/), which powers Linkerd's proxy.

We're in the process of migrating the proxy to Tokio 0.2 to take advantage of Rust's new `async`/`await` syntax, but following that, I think we'd be pretty keen to start building a plan to support proxy_wasm.

There's nothing in our [immediate roadmap](https://github.com/linkerd/linkerd2/discussions/4383) that really requires this level of flexibility, but ideally we'd have done a sufficient amount of prototyping and discovery over the next few months so that we can have a concrete plan to deliver this integration after `stable-2.10`.

@andrewdavidmackenzie Is this something you want to help contribute to the project? Or just checking our roadmap more generally? Either is fine, but if you're interested, we'd be more than willing to start talking through the plan in more detail!

**Comment by andrewdavidmackenzie** [comment]
Thanks for reply. Good to hear you're looking into it. I think it's something good for the overall ecosystem, as people developing filters and plugins are not committing themselves to just one proxy and they can re-use them across their different use cases possibly with different proxies.

I'm afraid I lack the knowledge and time to contribute such a thing. Our interest is in knowing that investments we make in writing proxy_wasm plugins are not tying us to just one proxy, and maintaining flexibility of what we can use in the "data plane".

thanks!

**Comment by sonhal** [comment]
Hi @olix0r, is there any new movement on this topic? Are you looking for contributors on this?

**Reply by olix0r** [reply]
@sonhal Sorry, I missed this message. There hasn't been any real movement. I think we'd pretty into contributions in this space. The key will be to make this a compile-time option so we can be incremental about introducing it into the system while we get comfortable with it.
